From ed55f639971b6ea44581b40402741807cbaf820f Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Tue, 15 Oct 2013 16:29:09 +0200
Subject: [PATCH 1/2] Some fixes in gtk_tree_view_is_blank_at_pos()

- Set path & column to NULL if blank
- _gtk_tree_view_column_is_blank_at_pos() would return FALSE instead of TRUE
  when no renderer was found at the given position.
- Fixed how we determine the expander column, which didn't account for using the
  first visible column

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 gtk/gtktreeview.c       | 14 ++++++++++----
 gtk/gtktreeviewcolumn.c |  2 +-
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/gtk/gtktreeview.c b/gtk/gtktreeview.c
index 5e1f476..487dbdf 100644
--- a/gtk/gtktreeview.c
+++ b/gtk/gtktreeview.c
@@ -14184,8 +14184,14 @@ gtk_tree_view_is_blank_at_pos (GtkTreeView       *tree_view,
   if (!gtk_tree_view_get_path_at_pos (tree_view, x, y,
                                       &real_path, &real_column,
                                       cell_x, cell_y))
-    /* If there's no path here, it is blank */
-    return TRUE;
+    {
+      /* If there's no path here, it is blank */
+      if (path)
+        *path = NULL;
+      if (column)
+        *column = NULL;
+      return TRUE;
+    }
 
   if (path)
     *path = real_path;
@@ -14197,8 +14203,8 @@ gtk_tree_view_is_blank_at_pos (GtkTreeView       *tree_view,
   _gtk_tree_view_find_node (tree_view, real_path, &tree, &node);
 
   /* Check if there's an expander arrow at (x, y) */
-  if (real_column == tree_view->priv->expander_column
-      && gtk_tree_view_draw_expanders (tree_view))
+  if (gtk_tree_view_draw_expanders (tree_view)
+      && gtk_tree_view_is_expander_column (tree_view, real_column))
     {
       gboolean over_arrow;
 
diff --git a/gtk/gtktreeviewcolumn.c b/gtk/gtktreeviewcolumn.c
index 9e1cba8..a8ff3bc 100644
--- a/gtk/gtktreeviewcolumn.c
+++ b/gtk/gtktreeviewcolumn.c
@@ -1553,7 +1553,7 @@ _gtk_tree_view_column_is_blank_at_pos (GtkTreeViewColumn *column,
                                                  background_area,
                                                  x, y);
   if (!match)
-    return FALSE;
+    return TRUE;
 
   gtk_cell_area_get_cell_allocation (priv->cell_area,
                                      priv->cell_area_context,
-- 
2.4.0

